tests/variables.test.vim

```diff
@@ -591,3 +591,67 @@ function! Test_EvaluateFailure()
   call vimspector#test#setup#Reset()
   %bwipe!
 endfunction
+
+function! Test_VariableEval()
+  let fn =  'testdata/cpp/simple/struct.cpp'
+  call s:StartDebugging( #{ fn: fn, line: 24, col: 1, launch: #{
+        \   configuration: 'run-to-breakpoint'
+        \ } } )
+
+  call vimspector#StepOver()
+  call vimspector#test#signs#AssertCursorIsAtLineInBuffer( fn, 26, 1 )
+
+  "evaluate the prev line
+  " call win_gotoid( g:vimspector_session_windows.code )
+  call setpos('.', [ 0, 24, 8 ])
+  call vimspector#test#signs#AssertCursorIsAtLineInBuffer( fn, 24, 8 )
+  call vimspector#ShowTooltip()
+
+  call WaitForAssert( {->
+        \   AssertMatchist(
+        \     [
+        \       '{...}',
+        \       ' - i: 0',
+        \       ' - c: 0 ''\\0\{1,3}''',
+        \       ' - fffff: 0',
+        \       ' + another_test: ',
+        \     ],
+        \     getbufline( winbufnr( g:vimspector_session_windows.eval ),
+        \                 1,
+        \                 '$' )
+        \   )
+        \ } )
+
+  " Expand
+  call win_execute( g:vimspector_session_windows.eval, [
+    \ 'call feedkeys( ''jjjj'', ''xt'' )',
+    \ 'call feedkeys( "\<CR>", ''xt'' )'
+    \ ] )
+
+  call WaitForAssert( {->
+        \   AssertMatchist(
+        \     [
+        \       '{...}',
+        \       ' - i: 0',
+        \       ' - c: 0 ''\\0\{1,3}''',
+        \       ' - fffff: 0',
+        \       ' - another_test: ',
+        \       '   - choo: 0 ''\\0\{1,3}''',
+        \       '   + ints: '
+        \     ],
+        \     getbufline( winbufnr( g:vimspector_session_windows.eval ),
+        \                 1,
+        \                 '$' )
+        \   )
+        \ } )
+
+  "Close
+  call win_execute( g:vimspector_session_windows.eval, 'call feedkeys( "\<Esc>", ''xt'')')
```

Comment:
for some reason i am not able to feed `Esc` key to the popup window. it gives `E994` error (operation is not allowed). i tried using a regular `x` to close the window, but that seems to have the same effect. any other key works just fine, so it seems the problem is tied specifically to the keys that are supposed to close the window.

any clue why this happening?

Reply:
i read you article, but somehow i did not put 2 and 2 together about feedkeys modifying the global typehead. anyway, the problem seems to be fixed once i remove `win_execute` call. still, i wonder what function is called by `popup_filter_menu` that would trigger this error on `esc`